### PR TITLE
Turn on clang-tidy check for function naming and change function names to pass check

### PR DESCRIPTION
**Pull request by helenarichie**
For some reason that's not currently clear to me, clang-tidy only catches some instances of function names not conforming to the `Upper_Snake_Case` naming convention that we've chosen. I've corrected all of the functions that the check was failing for (and others that I happened to catch while doing this) and the check now passes. 

For the most part, it was straightforward to choose new versions of the function names to conform to our naming standard, but there are a few cases that I could use a second opinion on. In particular, there were a few function names in models/disk_ICs.cpp that I didn't understand the meaning of, so the new names that I gave them might not be the best.

### CODE REVIEW

**Comment by bcaddy** [comment]
All the MHD and testing stuff looks good to me.

File: src/gravity/paris/ParisPeriodic.cu
Comment (ojwg) [inline src/gravity/paris/ParisPeriodic.cu:19]: `sqrt` is a library call, so this name change will break the code.

File: src/particles/feedback_CIC_gpu.cu
Comment (ojwg) [inline src/particles/feedback_CIC_gpu.cu]: HIP has `atomicMax`, so if compiling with O_HIP, we use it.  Changing this will break on AMD if you also update client code to call the new name.

**Review by ojwg — CHANGES_REQUESTED** [review]
I think the disk_ICs function renaming is good, but there are a couple library function names that need to be changed back.

File: src/particles/feedback_CIC_gpu.cu
Comment (helenarichie) [reply src/particles/feedback_CIC_gpu.cu]: Resolved in [1465895](https://github.com/helenarichie/cholla/commit/146589509a5250e3280c3212889c52a31ed513c1)

File: src/gravity/paris/ParisPeriodic.cu
Comment (helenarichie) [reply src/gravity/paris/ParisPeriodic.cu:19]: Maybe I'm misunderstanding something, but I thought those were calls to an inline function defined in [line 7](https://github.com/cholla-hydro/cholla/blob/5bdd42b8b728e4fdc39bfc783a3d38b4e88e239a/src/gravity/paris/ParisPeriodic.cu#L7) of this file that returns the square of the argument?

**Comment by evaneschneider** [comment]
I think this one is good to go once it's merged with the updated version of dev.

**Comment by helenarichie** [comment]
Okay, this is ready to go. I resolved the merge conflicts ([42fdb67](https://github.com/cholla-hydro/cholla/pull/310/commits/42fdb670c15b550188716d4a554484ac31acafaa)) and updated the function calls/new definitions to pass the function naming check ([2c0b6e2](https://github.com/cholla-hydro/cholla/pull/310/commits/2c0b6e2cb3d45af1557b6d2ffb4a45703e6d8ea3)). I also had to make a few extra commits for formatting and updating my test data submodule.

**Pull request merged**
